osu.Game/Screens/OnlinePlay/Match/RoomSubScreen.cs

```diff
@@ -374,10 +375,16 @@ private void selectedItemChanged()
             }
         }
 
-        private void beatmapUpdated(BeatmapSetInfo set) => Schedule(updateWorkingBeatmap);
+        private void beatmapUpdated(BeatmapSetInfo set) => Schedule(UpdateWorkingBeatmap);
 
-        private void updateWorkingBeatmap()
+        protected virtual void UpdateWorkingBeatmap()
         {
+            if (BeatmapAvailability.Value.State != DownloadState.LocallyAvailable)
+            {
+                Beatmap.Value = beatmapManager.GetWorkingBeatmap(null);
```

Comment:
Was this logic somewhere before this PR?

Reply:
See comment below.